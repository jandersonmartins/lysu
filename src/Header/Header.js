import React from 'react'
import { Link } from '../Common'
import * as S from './Header.styles'

const Header = () => (
  <S.Header>
    <S.H1>
      <Link href="/">Lysu</Link>
    </S.H1>
  </S.Header>
)

export default Header
