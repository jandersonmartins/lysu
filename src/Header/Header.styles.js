import styled from 'styled-components'

const Header = styled.header`
  height: 5rem;
  border-bottom: .1rem solid rgba(216, 216, 216, .7);
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: ${props => props.theme.colors.primary};
`

const H1 = styled.h1`
  font-size: 2rem;
`

export { Header, H1 }
