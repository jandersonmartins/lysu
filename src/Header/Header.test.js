import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import Header from './Header'

test('renders correcty', () => {
  const { container } = render(
    <ThemeProvider theme={{ colors: {} }}>
      <Header currentLang="en-US" onSelectLang={jest.fn()} />
    </ThemeProvider>
  )

  expect(container.firstChild).toMatchSnapshot()
})
