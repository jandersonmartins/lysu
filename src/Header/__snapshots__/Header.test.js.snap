// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correcty 1`] = `
.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2:link,
.c2:visited {
  font-size: inherit;
  font-weight: inherit;
  color: inherit;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
}

.c5 {
  cursor: pointer;
  font-size: 1.3rem;
}

.c0 {
  position: fixed;
  width: 100vw;
  height: 5rem;
  border-bottom: .1rem solid;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  font-size: 2rem;
}

.c3 {
  position: absolute;
  right: 2rem;
}

<header
  class="c0"
>
  <h1
    class="c1"
  >
    <a
      class="c2"
      href="/"
    >
      Lysu
    </a>
  </h1>
  <div
    class="c3"
  >
    <div
      class="c4"
    >
      <a
        class="c5"
        data-testid="current-lang"
      >
        EN
      </a>
    </div>
  </div>
</header>
`;
